src/http/router/token_hash_map.zig

```diff
@@ -0,0 +1,203 @@
+const std = @import("std");
+const Token = @import("routing_trie.zig").Token;
+
+/// Errors of get function.
+pub const MapGetErrors = error {
+    NotFound,
+};
+
+/// Type of a token hash.
+pub const Hash = u64;
+
+/// Type of a hash entry in hashed array.
+pub const HashEntry = struct{Hash, usize};
+
+/// In-place sort of the given array at compile time.
+/// Implementation reference: https://github.com/Koura/algorithms/blob/b1dd07147a34554543994b2c033fae64a2202933/sorting/quicksort.zig
+fn sort(A: []HashEntry, lo: usize, hi: usize) void {
+    if (lo < hi) {
+        const p = partition(A, lo, hi);
+        sort(A, lo, @min(p, p -% 1));
+        sort(A, p + 1, hi);
+    }
+}
+
+fn partition(A: []HashEntry, lo: usize, hi: usize) usize {
+    // Pivot can be chosen otherwise, for example try picking the first or random
+    // and check in which way that affects the performance of the sorting.
+    const pivot = A[hi][0];
+    var i = lo;
+    var j = lo;
+    while (j < hi) : (j += 1) {
+        if (A[j][0] <= pivot) {
+            std.mem.swap(HashEntry, &A[i], &A[j]);
+            i += 1;
+        }
+    }
+    std.mem.swap(HashEntry, &A[i], &A[hi]);
+    return i;
+}
+
+/// Compile-time token hash map.
+pub fn TokenHashMap(V: type) type {
+    return struct {
+        const Self = @This();
+
+        /// Type of a key-value tuple.
+        pub const KV = struct {
+            Token,
+            V,
+        };
+
+        /// Sorted array of tokens keys hashes.
+        /// Associate a key hash to an index in keys / values array.
+        hashes: []const HashEntry,
+
+        /// Keys of the map.
+        keys: []const Token,
+
+        /// Values of the map.
+        values: []const V,
+
+        /// Hash the given token key.
+        fn hashKey(input: Token) Hash {
```

Comment:
This is a general note since I don't want to put this everywhere but for zzz, We use `snake_case` for all of our methods like `serve_fs_dir` so these should probably also be `snake_case`.